align_transcriptions.py

```diff
@@ -0,0 +1,559 @@
+        upper_edge_anchor_aplit_idx_on_ref = find_nearest_idx(
+            ref_snippet_split_points, index_mapping_of_ref_to_qry[-1]
+        )
+        upper_ref_split_point_idx = find_matching_edge(
+            upper_edge_anchor_aplit_idx_on_ref,
+            upper_edge_hist,
+            ref_text,
+            ref_snippet_split_points,
+            char_dict,
+            dict_weights,
+        )
+    # Fallback when the query text cannot produce a valid histogram
+    else:
+        upper_ref_split_point_idx = len(ref_snippet_split_points) - 1
+
+    return (
+        ref_snippet_split_points[lower_ref_split_point_idx],
+        # Include the last splitted text
+        ref_snippet_split_points[
+            min(upper_ref_split_point_idx + 1, len(ref_snippet_split_points) - 1)
+        ],
+    )
+
+
+def align_split(
+    args: dict,
+    reference_text: str,
+    reference_time_index: np.ndarray,
+    text_natural_split_points: np.ndarray,
+    char_dict: dict[str, int],
+    char_dict_weights: np.ndarray,
+    idx: int,
+    split: dict,
+    trans_store: dict,
+) -> dict:
+    if args.transcripts_split_id is not None:
+        if idx != args.transcripts_split_id:
+            return
+
+    time_start_s, time_end_s = split
+    split_transcript = None
+
+    # TODO: The list structure for lookup by a sub prop of a json element
+    # Is inefficient - Consider storing the transcription store as an id: val lookup dict
+    # The reason we need to lookup by the split idx is that some splits would not have a transcription
+    for t_s in trans_store["transcripts"]:
+        if t_s["split_idx"] == idx:
+            split_transcript = t_s
+            break
+
+    # A missing transcript is possible, if for example
+    # The transcription server could not find any sppeach parts
+    # In the audio split. Skip this transcription - this sample
+    # Will be dropped from the output dataset
+    if split_transcript is None:
+        return None
+
+    split_seg_texts = []
+    max_logprob = -np.inf
+
+    for seg in split_transcript["segments"]:
+        split_seg_texts.append(seg["text"].strip())
+        max_logprob = max(max_logprob, seg["avg_logprob"])
+    split_qry_text = " ".join(split_seg_texts).strip()
+
+    is_fallback = False
+    try:
+        ref_text_around_split = re.sub(
+            r"\s+",
+            " ",
+            get_transcript_for_time_range(
+                reference_text,
+                reference_time_index,
+                text_natural_split_points,
+                time_start_s,
+                time_end_s,
+                reference_text_neighborhood_size=args.reference_text_neighborhood_size,
+            ),
+        )
+    except:
+        # In some cases - ref does not contain the ranges this split
+        # points at - fallback to the query as a last resort
+        ref_text_around_split = ""
+        is_fallback = True
+
+    if not is_fallback:
+        qury_embedding = encode_text_stream(
```

Comment:
qury -> query (if that's what you meant)

Reply:
Indeed - tried to use shorter name here - but saving one char looks like a typo indeed.